lib/matplotlib/axis.py

```diff
@@ -1515,37 +1515,101 @@ def set_major_formatter(self, formatter):
         """
         Set the formatter of the major ticker.
 
+        In addition to a `~matplotlib.ticker.Formatter` instance,
+        this also accepts a ``str`` or function.
+
+        For a ``str`` a `~matplotlib.ticker.StrMethodFormatter` is used.
+        The field used for the value must be labeled ``'x'`` and the field used
```

Comment:
I understand you're using field to be more general, but I'd go a drop more technical here 'cause I think it can be clearer:
``` 
the tick value argument must be named ``'x'`` and the tick position argument must be labeled ``'pos'``
```

Reply:
"Field" is the term used by the upstream Python documentation, so I think we should stick with that.  It really isn't an "argument".